cdap-ui/app/directives/metric-picker/metric-picker.js

```diff
@@ -109,45 +145,54 @@ angular.module(PKG.name + '.commons')
         elem.find('button').on('blur', onBlurHandler);
         elem.find('input').on('blur', onBlurHandler);
 
-        scope.$watchCollection('metric', function (newVal, oldVal) {
+        var metricChanged = function (newVal, oldVal) {
           ngModel.$validate();
 
           if(newVal.type !== oldVal.type) {
             scope.metric.context = '';
-            scope.metric.name = '';
+            scope.metric.resetNames();
```

Comment:
What happens in this scenario,

- Choose context 1 
- Add 5 metrics
- Erase context 1 and choose context 2

Will the 5 metrics be there but their values cleared out or are we removing all the metrics except for the one (starting back from default)?